pkg/airgap/update.go

```diff
@@ -318,18 +318,20 @@ func GetMissingRequiredVersions(app *apptypes.App, airgap *kotsv1beta1.Airgap) (
 		return nil, errors.Wrap(err, "failed to load license")
 	}
 
-	return getMissingRequiredVersions(airgap, license, appVersions.AllVersions)
+	return getMissingRequiredVersions(airgap, license, appVersions)
 }
 
-func getMissingRequiredVersions(airgap *kotsv1beta1.Airgap, license *kotsv1beta1.License, installedVersions []*downstreamtypes.DownstreamVersion) ([]string, error) {
+func getMissingRequiredVersions(airgap *kotsv1beta1.Airgap, license *kotsv1beta1.License, installedVersions *downstreamtypes.DownstreamVersions) ([]string, error) {
 	missingVersions := make([]string, 0)
-	if len(installedVersions) == 0 {
+	// If no versions are installed, we can consider this an initial install.
+	// If the current installed version is from a different channel, we can consider this an initial install.
+	if len(installedVersions.AllVersions) == 0 || (installedVersions.CurrentVersion != nil && installedVersions.CurrentVersion.ChannelID != airgap.Spec.ChannelID) {
```

Comment:
In order to have the same behavior as online mode, I think you should check the `app.ChannelChanged` field. The one additional thing you'd want to do is reset this field like [this](https://github.com/replicatedhq/kots/blob/f4004f71a21512d47726d43a1b79939d7324efe6/pkg/updatechecker/updatechecker.go#L341-L343) but for airgap [here](https://github.com/replicatedhq/kots/blob/f4004f71a21512d47726d43a1b79939d7324efe6/pkg/airgap/update.go#L209)